src/encoded/static/components/matrix_chip_seq.js

```diff
@@ -0,0 +1,906 @@
+    if (!context || !context.matrix || !context.matrix.x || !context.matrix.y || !assayTitle || !organismName) {
+        return null;
+    }
+
+    const subTabSource = 'biosample_ontology.classification';
+    const subTabs = context.matrix.x[subTabSource].buckets.map(x => x.key);
+    const chIPSeqData = {};
+
+    subTabs.forEach((subTab) => {
+        const xGroupBy1 = context.matrix.x.group_by[0];
+        const xGroupBy2 = context.matrix.x.group_by[1];
+        const headerRow = context.matrix.x[xGroupBy1].buckets.find(f => f.key === subTab)[xGroupBy2]
+            .buckets
+            .reduce((a, b) => {
+                const m = a.concat(b);
+                return m;
+            }, [])
+            .map(x => x.key);
+        const headerRowIndex = headerRow.reduce((x, y, z) => { x[y] = z; return x; }, []);
+        const headerRowLength = headerRow.length;
+        const yGroupBy1 = context.matrix.y.group_by[0];
+        const yGroupBy2 = context.matrix.y.group_by[1];
+
+        const yData = context.matrix.y[yGroupBy1].buckets
+            .find(rBucket => rBucket.key === organismName)[yGroupBy2].buckets
+            .reduce((a, b) => {
+                const m = {};
+                m[b.key] = b[xGroupBy1].buckets
+                    .filter(f => f.key === subTab)
+                    .reduce((x, y) => {
+                        x.push(y[xGroupBy2].buckets
+                            .reduce((i, j) => i.concat(j), []));
+                        return x;
+                    }, []);
+                return a.concat(m);
+            }, []);
+
+        const dataRowT = {};
+
+        yData.forEach((y) => {
+            const yKey = Object.keys(y)[0];
+            dataRowT[yKey] = dataRowT[yKey] || [...Array(headerRowLength + 1)].fill(0);
+            dataRowT[yKey][0] = yKey;
+
+            const keyDocCountPair = y[yKey].reduce((a, b) => {
+                const m = a.concat(b);
+                return m;
+            }, []);
+
+            keyDocCountPair.forEach((kp) => {
+                const key = kp.key;
+                const docCount = kp.doc_count;
+                const index = headerRowIndex[key];
+                dataRowT[yKey][index + 1] = docCount;
+            });
+        });
+
+        let dataRow = [];
+        const keys = Object.keys(dataRowT);
+
+        // move biosample ontology classifications to dataRow-group
+        keys.forEach((key) => {
+            dataRow.push(dataRowT[key]);
+        });
+
+        // Remove all rows with no data (all entries/count are zero)
+        // Remove this line if you want to show all data and not just ones with content
+        dataRow = dataRow.filter((d) => {
+            const data = [...d]; // !Imporant. Clone to avoid modifying dataRow itself
+            data.shift(); // remove first entry as it is biosample ontology classification and not a count
+            return !data.every(i => i === 0); // remove all rows with all zeros
+        });
```

Comment:
Done, kind of. I made this more tense. I made it verbose initially for others to understand each step. I changed this to be more tense. I used some instead of every as it fast.